velox/functions/sparksql/RegexFunctions.cpp

```diff
@@ -135,79 +127,101 @@ struct RegexpReplaceFunction {
       const arg_type<Varchar>& pattern,
       const arg_type<Varchar>& replace,
       const arg_type<int64_t>& position) {
-    VELOX_USER_CHECK_GE(position, 1, "regexp_replace requires a position >= 1");
+    if (performChecks(result, stringInput, pattern, replace, position - 1)) {
```

Comment:
A conventional way is to define a template method with bool isAscii template parameter, then use `functions::stringImpl::cappedUnicode<isAscii>`.